### PR TITLE
Make rubber-banding distances different between AI karts

### PR DESCRIPTION
**Pull request by Alayan-stk-2**
See #3222 for the reasoning behind this PR, which aims to fix both points.

I've also made the AI slow down before it even reaches the player in the easiest difficulty and increased the distance range over which the changes happens (for reference, math class has 500 of length, so the 30 to 60 range from full-speed to min-speed which existed before was definitely tiny)

The only slight issue I see with the change is that using the current AI position as a base may tend to make light karts (who have a start bonus) better ranked and heavy AIs worse ranked. Other possibilities exist, like using the kart's world ID, though it may things slightly easier.

### CODE REVIEW

File: src/karts/controller/skidding_ai.cpp [hunk 1]
```diff
@@ -1965,23 +1972,69 @@ void SkiddingAI::computeNearestKarts()
...
-        unsigned int kart_id =
-            m_world->getPlayerKart(i)->getWorldKartId();
-        if(m_world->getOverallDistance(kart_id)>max_overall_distance)
-            max_overall_distance = m_world->getOverallDistance(kart_id);
+        unsigned int kart_id = m_world->getPlayerKart(i)->getWorldKartId();
+        overall_distance.push_back(m_world->getOverallDistance(kart_id));
+    }
+
+    // Sort the list (the list is very short, so a basic bubble sort suffice)
+    for(unsigned int i=0; i<n-1; i++)
```
Comment (auriamg) [inline src/karts/controller/skidding_ai.cpp]: Even so, please don't reinvent the wheel. Just use one of the standard sorting methods

File: src/karts/controller/test_ai.cpp [hunk 2]
```diff
@@ -315,7 +315,7 @@ void SkiddingAI::update(int ticks)
     computeNearestKarts();
 
     m_kart->setSlowdown(MaxSpeed::MS_DECREASE_AI,
-                        m_ai_properties->getSpeedCap(m_distance_to_player),
+                        m_ai_properties->getSpeedCap(m_distance_to_player, 2, 3),
```
Comment (auriamg) [inline src/karts/controller/test_ai.cpp]: Why are 2,3 hardcoded?

Comment (Alayan-stk-2) [reply src/karts/controller/test_ai.cpp]: It returns the average of the first speed cap and the last speed cap. I could have put 1,1 as well, though, as the test_ai is only used in profile_mode where there will be no slowdown anyway.

Plus, the test_ai is supposed to be updated with the new skidding_ai once changes have been validated (it has not been done recently, but it should be), at which point this will be overwritten by the changes seen in skidding_ai.

Basically, it doesn't matter and the main motivation behind changing anything at all in this file is avoiding a compilation error.

Comment (auriamg) [reply src/karts/controller/test_ai.cpp]: A little comment explaining that would be welcome. something like 1,1 /* N/A, no rubberband in profile mode */);
